Lib/fontTools/varLib/merger.py

```diff
@@ -60,6 +60,11 @@ def mergersFor(celf, thing, _default={}):
 		return _default
 
 	def mergeObjects(self, out, lst, exclude=()):
+		if hasattr(out, "ensureDecompiled"):
+			out.ensureDecompiled()
+		for item in lst:
+			if hasattr(item, "ensureDecompiled"):
+				item.ensureDecompiled()
 		keys = sorted(vars(out).keys())
```

Comment:
@justvanrossum I see why we need to do `ensureDecompiled`. It's because we call `vars(out)`, but the latter probably only gets the object's `__dict__` object, so `__getattr__` method is not triggered and the lazy decompilation doesn't occur.